ext/web/lib.deno_web.d.ts

```diff
@@ -577,66 +592,62 @@ declare var File: {
 };
 
 /** @category Streams */
-declare interface ReadableStreamDefaultReadDoneResult {
+declare type ReadableStreamReader<T> =
+  | ReadableStreamDefaultReader<T>
+  | ReadableStreamBYOBReader;
+
+/** @category Streams */
+declare type ReadableStreamController<T> =
+  | ReadableStreamDefaultController<T>
+  | ReadableByteStreamController;
+
+/** @category Streams */
+declare interface ReadableStreamGenericReader {
+  readonly closed: Promise<undefined>;
+  cancel(reason?: any): Promise<void>;
+}
+
+/** @category Streams */
+declare interface ReadableStreamReadDoneResult<T> {
   done: true;
-  value?: undefined;
+  value?: T;
 }
 
 /** @category Streams */
-declare interface ReadableStreamDefaultReadValueResult<T> {
+declare interface ReadableStreamReadValueResult<T> {
   done: false;
   value: T;
 }
 
 /** @category Streams */
-declare type ReadableStreamDefaultReadResult<T> =
-  | ReadableStreamDefaultReadValueResult<T>
-  | ReadableStreamDefaultReadDoneResult;
+declare type ReadableStreamReadResult<T> =
+  | ReadableStreamReadValueResult<T>
+  | ReadableStreamReadDoneResult<T>;
 
 /** @category Streams */
-declare interface ReadableStreamDefaultReader<R = any> {
-  readonly closed: Promise<void>;
-  cancel(reason?: any): Promise<void>;
-  read(): Promise<ReadableStreamDefaultReadResult<R>>;
+declare interface ReadableStreamDefaultReader<R = any>
+  extends ReadableStreamGenericReader {
+  read(): Promise<ReadableStreamReadResult<R>>;
   releaseLock(): void;
 }
 
 /** @category Streams */
 declare var ReadableStreamDefaultReader: {
   readonly prototype: ReadableStreamDefaultReader;
-  new <R>(stream: ReadableStream<R>): ReadableStreamDefaultReader<R>;
+  new <R = any>(stream: ReadableStream<R>): ReadableStreamDefaultReader<R>;
 };
 
-/** @category Streams */
-declare interface ReadableStreamBYOBReadDoneResult<V extends ArrayBufferView> {
-  done: true;
-  value?: V;
-}
-
-/** @category Streams */
-declare interface ReadableStreamBYOBReadValueResult<V extends ArrayBufferView> {
-  done: false;
-  value: V;
-}
-
-/** @category Streams */
-declare type ReadableStreamBYOBReadResult<V extends ArrayBufferView> =
-  | ReadableStreamBYOBReadDoneResult<V>
-  | ReadableStreamBYOBReadValueResult<V>;
```

Comment:
It can be reland when supported by upstream (https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1676)